assignments/this.js

```diff
@@ -1,26 +1,58 @@
 /* The for principles of "this";
-* in your own words. explain the four principle for the "this" keyword below.
-*
-* 1. 
-* 2. 
-* 3. 
-* 4. 
-*
-* write out a code example of each explanation above
-*/
+ * in your own words. explain the four principle for the "this" keyword below.
+ *
+ * 1. Window Binding is also known is global binding. 'this' refers to the global (aka window) object
+ * 2. Implicit Binding points to the object on which the function is called. In other words, it points to what's to the left of the period when the function is called
+ * 3. New Binding uses the 'new' keyword and constructs a new object. 'this' points to what is being constructed
+ * 4. With Explicit Binding, we can tell the Javascript engine to make 'this' point to a certain value using call(), apply(), or bind()
+ *
+ * write out a code example of each explanation above
+ */
 
 // Principle 1
-
 // code example for Window Binding
+function globalBinding() {
+    console.log(this); //points to the window object
+}
+// globalBinding();
 
 // Principle 2
-
 // code example for Implicit Binding
+function logDrake() {
+    console.log(this.yuh);
+}
 
-// Principle 3
+let drake = {
+    name: 'Drake',
+    yuh: 'yuh',
+    logDrake: logDrake
+}
+
+drake.logDrake();
 
+// Principle 3
 // code example for New Binding
+function Skateboard(brand) {
```

Comment:
no "this" keyword being used.. here's an example of using the New Binding (constructor):

// Principle 3

// code example for New Binding
// Constructor Function:
function Rockstar(name, instrument, band) {
    this.greeting = "What's Up?!";
    this.name = name;
    this.instrument = instrument;
    this.band = band;
    this.introduceMyself = function () {
        console.log(`${this.greeting}, I'm ${this.name} and I play ${this.instrument} for the band ${this.band}.`)
    }
}
    const slash = new Rockstar('Slash', 'Guitar', 'Guns N Roses');
    const tBarker = new Rockstar('Travis Barker', 'Drums', 'Blink-182');
    const flea = new Rockstar('Flea', 'Bass', 'The Red Hot Chili Peppers');
    slash.introduceMyself();
    tBarker.introduceMyself();
    flea.introduceMyself();